portal_tables/sync_publications.py

```diff
@@ -34,81 +41,113 @@ def add_missing_info(pubs, grants):
     pubs.loc[:, 'Link'] = [
         "".join(["[PMID:", str(pmid), "](", url, ")"])
         for pmid, url
-        in zip(pubs.pubmedId, pubs.pubmedUrl)
+        in zip(pubs['Pubmed Id'], pubs['Pubmed Url'])
     ]
-    pubs['grantName'] = ""
-    for i, row in pubs.iterrows():
-        grant_names = []
-        for g in row['publicationGrantNumber']:
-            grant_names.append(
-                grants[grants.grantNumber == g]['grantName'].values[0])
-        pubs.at[i, 'grantName'] = grant_names
+    
+    pattern = re.compile("(\')([\s\w/-]+)(\')")
+    
+    for col in new_cols: 
+        pubs[col] = ""
+        for row in pubs.itertuples():
+            i = row[0]
+            n = row[4].split(",")
+            extracted = []
+            for g in n:
+                if len(grants[grants.grantNumber == g][col].values) > 0:
+                    values = str(grants[grants.grantNumber == g][col].values[0])
+                    
+                    if col == 'grantName':
+                        extracted.append(values)
+                    
+                    else:
+                        matches = pattern.findall(values)
+                        for m in matches:
+                            extracted.append(m[1])
+
+                else:
+                    print(f"No match found for grant number: {g}")
+                    continue
+            
+            clean_values = list(dict.fromkeys(extracted))
+
+            pubs.at[i, col] = clean_values
+    
     return pubs
 
 
-def sync_table(syn, pubs, table):
+def sync_table(syn, pubs, table, dryrun):
     """Add pubs annotations to the Synapse table."""
     schema = syn.get(table)
 
     # Reorder columns to match the table order.
     col_order = [
-        'publicationDoi', 'publicationJournal', 'pubmedId', 'pubmedUrl',
-        'Link', 'publicationTitle', 'publicationYear', 'publicationKeywords',
-        'publicationAuthors', 'publicationAbstract', 'publicationAssay', 
-        'publicationTumorType', 'publicationTissue', 'publicationThemeName',
-        'publicationConsortiumName', 'publicationGrantNumber', 'grantName',
-        'publicationDatasetAlias', 'publicationAccessibility'
+        'Publication Doi', 'Publication Journal', 'Pubmed Id', 'Pubmed Url',
+        'Link', 'Publication Title', 'Publication Year', 'Publication Keywords',
+        'Publication Authors', 'Publication Abstract', 'Publication Assay', 
+        'Publication Tumor Type', 'Publication Tissue', 'theme',
+        'consortium', 'Publication Grant Number', 'grantName',
+        'Publication Dataset Alias', 'Publication Accessibility',
+        'entityId'
     ]
     pubs = pubs[col_order]
 
-    # Convert list column into string to match with table schema.
-    pubs.loc[:, 'publicationDatasetAlias'] = (
-        pubs['publicationDatasetAlias']
-        .str.join(", ")
-    )
-
-    new_rows = pubs.values.tolist()
-    syn.store(Table(schema, new_rows))
+    if dryrun is False:
```

Comment:
nit: it's generally considered bad practice to compare directly with `True`/`False`. Instead of use:

`if variable:`

or 

`if not variable:`

to compare truthy/falsy values.  Doing this also executes faster!

Reply:
Good call! I had `if not dryrun` at first and then changed it without thinking, so thank you for noting this!